actions_helper/main.py

```diff
@@ -0,0 +1,112 @@
+import os
+from dataclasses import dataclass
+
+import boto3
+import click
+from botocore.client import BaseClient
+
+
+class NonSingleValueError(Exception):
+    pass
+
+
+@dataclass
+class Tag:
+    key: str
+    value: str
+
+
+def format_tags(task_definition_tags: str) -> tuple[Tag, ...]:
+    try:
+        return tuple(
+            Tag(tag.strip().split(":")[0], tag.strip().split(":")[1]) for tag in task_definition_tags.split(",")
+        )
+    except IndexError:
+        raise ValueError(
+            "Invalid task definition tag format, expects 'key:value,key:value'",
+        )
+
+
+def get_active_task_definition_arn_by_tag(
+    *,
+    ecs_client: BaseClient,
+    task_definition_family_prefix: str,
+    task_definition_tags: str,
+    allow_initial_deployment: bool,
+) -> str:
+    tags = format_tags(task_definition_tags)
+
+    active_task_definition_arns = ecs_client.list_task_definitions(
+        familyPrefix=task_definition_family_prefix,
+        status="ACTIVE",
+        sort="DESC",
+    )["taskDefinitionArns"]
+
+    active_task_definitions = [
+        task_definition["taskDefinition"]["taskDefinitionArn"]
+        for task_definition_arn in active_task_definition_arns
+        if (
+            task_definition := ecs_client.describe_task_definition(
+                taskDefinition=task_definition_arn,
+                include=["TAGS"],
+            )
+        )
```

Comment:
Can you please explain why you need the if condition here? To me, it implies that `describe_task_definition` can return None for the `task_definition_arn`s we are passing, which is unexpected as we just listed them from the active task definitions in the lines above. In which situation can this happen?

Reply:
Since we are using a list compression; as preferred; 

> A list comprehension consists of brackets containing an expression followed by a for clause, then zero or more for or if clauses. 

It needs to be done, to make use of the described task definition to assert tags as in `all({"key": tag.key, "value": tag.value} in task_definition["tags"] for tag in tags)`; 

It does not implies it can None

Reply:
Thank you for looking at this together with me. I understand now why it is done this way, and I also don't see any obvious way to improve this further.